### PR TITLE
StartupTasks are registered on the FeatureConfigurationContext

### PR DESCRIPTION
**Pull request by danielmarbach**
Implements #3060 

For :heart_eyes: @SzymonPobiega ;)

Related doco update

https://github.com/Particular/docs.particular.net/pull/953

@Particular/nservicebus Please review

### CODE REVIEW

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt [hunk 1]
```diff
@@ -1473,22 +1475,25 @@ namespace NServiceBus.Features
         public NServiceBus.Pipeline.PipelineSettings Pipeline { get; }
         public NServiceBus.Settings.ReadOnlySettings Settings { get; }
         public NServiceBus.Pipeline.PipelineSettings AddSatellitePipeline(string name, string qualifier, NServiceBus.Transports.TransactionSupport requiredTransactionSupport, NServiceBus.Transports.PushRuntimeSettings runtimeSettings, out string transportAddress) { }
+        public void RegisterStartupTask(NServiceBus.Features.FeatureStartupTask startupTask, string name) { }
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Do we need the name to be explicit? Could we default to the name of the class?

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Only if I introduce a generic ;(

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: @andreasohlund pushed an update

Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: :dancer:

**Comment by SzymonPobiega** [comment]
I've fixed all the ATs and made a small refactoring wrt how FST are started and stopped. Can somebody in @Particular/nservicebus review this?

**Pull request merged**